source/common/upstream/outlier_detection_impl.cc

```diff
@@ -207,10 +232,54 @@ void DetectorImpl::onConsecutive5xxWorker(HostPtr host) {
   ejectHost(host, EjectionType::Consecutive5xx);
 }
 
+double DetectorImpl::srEjectionThreshold(double sr_sum, std::vector<double>& sr_data) {
+  double mean = sr_sum / sr_data.size();
```

Comment:
Can you move this to a utility class and then have tests around it? And link to where you got the algorithm from.